src/lang/LanguageManager.php

```diff
@@ -0,0 +1,169 @@
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\lang;
+
+use function array_filter;
+use function array_map;
+use function explode;
+use function file_exists;
+use function is_dir;
+use function parse_ini_file;
+use function scandir;
+use function substr;
+use function is_array;
+use const INI_SCANNER_RAW;
+use const SCANDIR_SORT_NONE;
+
+class LanguageManager
+{
+	/** @var Language[] */
+	private $languages = [];
+
+	/** @var string */
+	public const FALLBACK_LANGUAGE = 'eng';
+
+	/** @var string[] */
+	public static $localeToLangCode = [
+		'bg_BG' => 'bul',
+		'cs_CZ' => 'ces',
+		'da_DK' => 'dan',
+		'de_DE' => 'deu',
+		'el_GR' => 'eli',
+		'en_GB' => 'eng',
+		'en_US' => 'eng',
+		'es_ES' => 'spa',
+		'es_MX' => 'spa',
+		'fi_FI' => 'fin',
+		'fr_CA' => 'fra',
+		'fr_FR' => 'fra',
+		'hu_HU' => 'hun',
+		'id_ID' => 'ind',
+		'it_IT' => 'ita',
+		'ja_JP' => 'jpn',
+		'ko_KR' => 'kor',
+		'nb_NO' => 'nor',
+		'nl_NL' => 'nld',
+		'pl_PL' => 'pol',
+		'pt_BR' => 'por',
+		'pt_PT' => 'por',
+		'ru_RU' => 'rus',
+		'sk_SK' => 'slv',
+		'sv_SE' => 'swe',
+		'tr_TR' => 'tur',
+		'uk_UA' => 'ukr',
+		'zh_CN' => 'chs',
+		'zh_TW' => 'zho'
+	];
+
+	public function __construct(string $path)
+	{
+		$this->loadLanguages($path);
+	}
+
+	/**
+	 * @return string[]
+	 * @phpstan-return array<string, string>
+	 *
+	 * @throws LanguageNotFoundException
+	 */
+	public static function getLanguageList(string $path = "") : array{
+		if($path === ""){
+			$path = \pocketmine\RESOURCE_PATH . "locale/";
+		}
+
+		if(is_dir($path)){
+			$allFiles = scandir($path, SCANDIR_SORT_NONE);
+
+			if($allFiles !== false){
+				$files = array_filter($allFiles, function(string $filename) : bool{
+					return substr($filename, -4) === ".ini";
+				});
+
+				$result = [];
+
+				foreach($files as $file){
+					$code = explode(".", $file)[0];
+					$strings = self::loadLang($path, $code);
+					if(isset($strings["language.name"])){
+						$result[$code] = $strings["language.name"];
+					}
+				}
+
+				return $result;
+			}
+		}
+
+		throw new LanguageNotFoundException("Language directory $path does not exist or is not a directory");
+	}
+
+	public function get(string $langCode): ?Language
+	{
+		return $this->languages[$langCode] ?? $this->languages[self::$localeToLangCode[$langCode]] ?? $this->languages[self::FALLBACK_LANGUAGE] ?? null;
+	}
+
+	public function loadLanguages(string $path = ''): void
+	{
```

Comment:
Formatting: brace indentation should match the rest of PocketMine's classes.

Reply:
This is not applicable you can fix it